### PR TITLE
chore: mark aws-cdk-lib as public and hypermodular packages as private

### PR DESCRIPTION
**Pull request by nija-at**
All hypermodular packages under the `@aws-cdk/` folder are marked as
private. `aws-cdk-lib` is marked as public.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/package.json [hunk 1]
```diff
@@ -61,12 +59,11 @@
         "maven": {
           "groupId": "software.amazon.awscdk",
           "artifactId": "lib",
```
Comment (njlynch) [inline packages/aws-cdk-lib/package.json:61]:
```suggestion
          "artifactId": "lib"
```

Think this is what's breaking the build.

**Review by njlynch — APPROVED** [review]

File: packages/aws-cdk-lib/package.json (hunk 1 above)
Comment (nija-at) [reply packages/aws-cdk-lib/package.json:61]: Gah!!

**Comment by NetaNir** [comment]
Note to self - look at the branch name. For a second there I thought my time travel machine works!

**Comment by eladb** [comment]
This would have been soooooooooo easy with projen :-(

**Review by eladb — CHANGES_REQUESTED** [review]
I was expecting to find some pkglint rule that enforces this...

**Comment by nija-at** [comment]
> I was expecting to find some pkglint rule that enforces this...

The rule is here - https://github.com/aws/aws-cdk/pull/11317/files. It's in a separate PR because it gets merged into master.

**Review by eladb — APPROVED** [review]

**Pull request merged**